src/ucm/cuda/cudamem.c

```diff
@@ -99,6 +99,32 @@ ucm_dispatch_mem_type_free(void *addr, size_t length, ucs_memory_type_t mem_type
     ucm_event_dispatch(UCM_EVENT_MEM_TYPE_FREE, &event);
 }
 
+static ucs_memory_type_t ucm_cuda_get_mem_type(void *ptr)
+{
+    struct cudaPointerAttributes ptr_attrs;
+    ucs_memory_type_t mem_type;
+    cudaError_t cu_err;
+
+    cu_err = cudaPointerGetAttributes(&ptr_attrs, ptr);
+    if (cu_err != cudaSuccess) {
+        return UCS_MEMORY_TYPE_LAST; /* unknown memory type */
```

Comment:
I think we can't return LAST here.  it must be some type of cuda memory if it is called cudaFree(). I think we should print error/warn and return UCS_MEMORY_TYPE_CUDA. WDYT?

Reply:
@bureddy thanks, good idea! Fixed